packages/dao/contracts/Api3Template.sol

```diff
@@ -59,46 +64,55 @@ contract Api3Template is Api3BaseTemplate {
             address(dao),
             address(acl),
             address(_api3Pool),
-            address(voting),
-            address(agent)
+            address(mainVoting),
+            address(secondaryVoting),
+            address(mainAgent),
+            address(secondaryAgent)
         );
     }
 
     function _setupApps(
         Kernel _dao,
         ACL _acl,
         MiniMeToken _api3Pool,
-        uint64[3] memory _votingSettings,
+        uint64[3] memory _mainVotingSettings,
+        uint64[3] memory _secondaryVotingSettings,
         address _permissionManager
     )
     internal
-    returns (Api3Voting, Agent)
+    returns (Api3Voting, Api3Voting, Agent, Agent)
     {
-        Agent agent = _installDefaultAgentApp(_dao);
-        Api3Voting voting = _installVotingApp(_dao, _api3Pool, _votingSettings);
+        (Agent mainAgent, Agent secondaryAgent) = _installDefaultAgentApps(_dao);
+        (Api3Voting mainVoting, Api3Voting secondaryVoting) = _installVotingApps(_dao, _api3Pool, _mainVotingSettings, _secondaryVotingSettings);
 
         _setupPermissions(
             _acl,
-            agent,
-            voting,
+            mainAgent,
+            mainVoting,
+            secondaryAgent,
+            secondaryVoting,
             _permissionManager
         );
 
-        return (voting, agent);
+        return (mainVoting, secondaryVoting, mainAgent, secondaryAgent);
     }
 
     function _setupPermissions(
         ACL _acl,
-        Agent _agent,
-        Api3Voting _voting,
+        Agent _mainAgent,
+        Api3Voting _mainVoting,
+        Agent _secondaryAgent,
+        Api3Voting _secondaryVoting,
         address _permissionManager
     )
     internal
     {
-        _createAgentPermissions(_acl, _agent, _voting, _permissionManager);
-        _createVaultPermissions(_acl, Vault(_agent), _voting, _permissionManager);
+        _createAgentPermissions(_acl, _mainAgent, _mainVoting, _permissionManager);
+        _createAgentPermissions(_acl, _secondaryAgent, _secondaryVoting, _permissionManager);
+        _createVaultPermissions(_acl, Vault(_mainAgent), _mainVoting, _permissionManager);
         _createEvmScriptsRegistryPermissions(_acl, _permissionManager, _permissionManager);
-        _createVotingPermissions(_acl, _voting, _voting, ANY_ENTITY, _permissionManager);
+        _createVotingPermissions(_acl, _mainVoting, _mainAgent, ANY_ENTITY, _permissionManager);
```

Comment:
Actually I checked and the voting app can own permissions over itself. Is this a Burak request ? Otherwise I think you can just put the voting apps as a parameter - instead of `_mainAgent`.

Reply:
Yep, at least as I understood Burak asked about it:

> Yeah so I assume that the DAO template should be modified to give the primary Agent MODIFY_SUPPORT_ROLE and MODIFY_QUORUM_ROLE for both Api3Voting apps?

Reply:
Maybe confirm with him. He may have understood the framework wrong. Because it introduces an un-necessary transaction step - thus UX issues, gas costs, etc.

Reply:
Yeah I know nothing about how this works, I assumed the voting app couldn't have permission directly and would need the agent in-between. You should always take whatever I say about Aragon with a grain of salt.

Reply:
Let's assign it to the very voting app, then.